jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/ChunksContentSource.java

```diff
@@ -73,7 +73,7 @@ public Content.Chunk read()
             if (last)
                 terminated = Content.Chunk.EOF;
         }
-        return Content.Chunk.from(chunk.getByteBuffer().slice(), chunk.isLast(), chunk);
```

Comment:
Yay, this was missing a `retain()` call!

Reply:
Agreed, I think just `return chunk` should be enough here.